### PR TITLE
[ENH] add preregistration URL to dataset description 

### PR DESCRIPTION
**Pull request by Remi-Gau**
This was mentioned in #552 and could constitute a small / easy addition to the dataset description.

- [x] It there is some interest I can add a line in the JSON example. 

to do:

- [x] open PR to bids-validator to add the field, see https://github.com/bids-standard/bids-specification/pull/572#issuecomment-732167563

### CODE REVIEW

**Review by sappelhoff — DISMISSED** [review]

**Review by CPernet — APPROVED** [review]

**Comment by emdupre** [comment]
We should probably add this field to the example JSON immediately below as well.

More generally, I suppose I'm a bit confused about this field. I usually see pre-registration to describe a specific effect, rather than collecting certain kinds of data. That is, we otherwise include optional fields that are focused on the data itself (like the data DOI rather than the paper DOI, or the ethics approval), rather than optional fields about the study the data come from.

Was there a conversation about what this field adds ? Sorry to be behind on this !

**Comment by Remi-Gau** [comment]
This was one of the suggestions I made in #552 and is somewhat related to #557, though I admit that there has not been a lengthy discussion about this. So you not behind in any way.

**Comment by Remi-Gau** [comment]
> I usually see pre-registration to describe a specific effect, rather than collecting certain kinds of data. The reason that's an important distinction for me here is that -- from my read -- we otherwise include optional fields that are focused on the data itself (like the data DOI rather than the paper DOI, or the ethics approval), rather than optional fields about the study where the data come from.

Well, unless I am mistaken, pre-registration should ideally describe (amongst other things) many of the decisions that will constrain what the data will be and how much of it there will be and hence what effect size one can hopefully expect to detect in this dataset in general and not just for the effect of interest.

Ideally, this should be taken care of by having info about the power analysis and sample size estimation (see #557) but more generally pre-reg contain a lot of potential metadata about the dataset that doesn't fit (yet but might never fit) into BIDS, and having a pointer to the pre-reg might be a way to "add" that extra information.

Also ethics approval and funding are already part of dataset_description.json so I think that it is not that clear if we should completely avoid items that are not "just on the data itself" but might also pertain to the study or set of studies the dataset is part of.

Though I agree that this could get us onto the slippery slopes of mount scope creep...    :mountain: :roller_coaster:

**Comment by emdupre** [comment]
> Also ethics approval and funding are already part of dataset_description.json so I think that it is not that clear if we should completely avoid items that are not "just on the data itself" but might also pertain to the study or set of studies the dataset is part of.

Personally, I see these as more directly related to the data itself, but that might just be because they're so much more common (or even necessary) to collect data, whereas pre-reg is still relatively rare in neuroimaging. 

> more generally pre-reg contain a lot of potential metadata about the dataset that doesn't fit (yet but might never fit) into BIDS, and having a pointer to the pre-reg might be a way to "add" that extra information.

I'm not against more information, for sure ! I'm fine with this if you think it will be useful :+1:

**Comment by Remi-Gau** [comment]
For more context, I think the idea partly came from a mix of reflection on the COBIDAS front and also some of the [guidelines for PET data](https://journals.sagepub.com/doi/10.1177/0271678X20905433) that are at the core of the PET BEP (though I don't think pre-reg is IN the BEP).

Maybe we should not "rush" on this one just because **I** think it will be useful.

Tagging @melanieganz who maybe has an opinion on this.

File: src/03-modality-agnostic-files.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Every dataset MUST include this file with the following fields:
 | EthicsApprovals    | OPTIONAL. List of ethics committee approvals of the research protocols and/or protocol identifiers.                                                                                                                                                                |
 | ReferencesAndLinks | OPTIONAL. List of references to publication that contain information on the dataset, or links.                                                                                                                                                                     |
 | DatasetDOI         | OPTIONAL. The Document Object Identifier of the dataset (not the corresponding paper).                                                                                                                                                                             |
+| Preregistration    | OPTIONAL.  URL or Document Object Identifier of the time-stamped pre-registration of the study that produced the dataset.                                                                                                                                          |
```
Comment (effigies) [inline src/03-modality-agnostic-files.md:31]: DOIs can be made URIs by prepending `doi:` (or URLs by prepending `https://doi.org/`), so this puts the address of one URI type on par with a full URI. An implementing tool would basically have to say:

```
try:
    val = parseurl(preregistration)
except:
    val = parseuri("doi:" + preregistration)
```

It seems conceptually cleaner to permit any URI:

```suggestion
| Preregistration    | OPTIONAL. URI of a time-stamped preregistration of the study that produced the dataset.                                                                                                                                                                            |
```

I'm not sure if there's an argument for narrowing to URL.

I am also :+1: for adding an example URI to add to the `dataset_description.json`.

Comment (CPernet) [reply src/03-modality-agnostic-files.md:31]: a `study that produced the dataset` sounds weird, people produce datasets
suggest
`URL or DOI of a time-stamped preregistration of the study from which the dataset originates.`

**Comment by Remi-Gau** [comment]
Looking through an [unofficial list of pre-reg](https://docs.google.com/spreadsheets/d/1_R3UBKvT4vEDOAPXxR8W98QGvZMhpuyxlvYzIGekf0w/edit?usp=sharing) I spotted some that could be interesting:
- pre-reg of th [NARPS dataset](https://aspredicted.org/ru8ua.pdf)
- an EEG [pre-reg](https://osf.io/uf9gh)
- an fMRI [pre-reg](https://osf.io/8hwdv)

Any preference? Or better suggestions?

**Comment by effigies** [comment]
No preference for prereg.

I think I would rather have the URI definition in the spec, rather than directing readers to the starter kit. That is supposed to be informative but not authoritative. I will post more to #601.

**Comment by effigies** [comment]
This needs to be rebased on master, and use the common URI definition.

**Comment by Remi-Gau** [comment]
Finally took 5 minutes to make those changes and add and a link to an example pre-reg...

**Comment by effigies** [comment]
Looks like the failing link is https://www.ncbi.nlm.nih.gov/pubmed/24474423 which redirects to https://pubmed.ncbi.nlm.nih.gov/24474423/. I guess we need to update PubMed links.

**Review by effigies — APPROVED** [review]

**Comment by Remi-Gau** [comment]
> Looks like the failing link is https://www.ncbi.nlm.nih.gov/pubmed/24474423 which redirects to https://pubmed.ncbi.nlm.nih.gov/24474423/. I guess we need to update PubMed links.

If you give me a :+1: I can smuggle that change into the PR, if you want. :wink:

**Comment by sappelhoff** [comment]
Why not do the link changes in the link PR we have open? #676 

:-)

**Review by sappelhoff — APPROVED** [review]

**Comment by sappelhoff** [comment]
oh @Remi-Gau could you please open a PR to add this field to bids-validator here: https://github.com/bids-standard/bids-validator/blob/master/bids-validator/validators/json/schemas/dataset_description.json

then we can merge the validator PR as soon as we merge this one.

All given that no-one objects with valid arguments.

File: src/03-modality-agnostic-files.md [hunk 2]
```diff
@@ -55,7 +56,8 @@ Example:
     "Alzheimer A., & Kraepelin, E. (2015). Neural correlates of presenile dementia in humans. Journal of Neuroscientific Data, 2, 234001. doi:1920.8/jndata.2015.7"
   ],
   "DatasetDOI": "doi:10.0.2.3/dfjj.10",
-  "HEDVersion": "7.1.1"
+  "HEDVersion": "7.1.1",
+  "Preregistration": "doi:10.1038/s41598-018-30701-5"
```
Comment (emdupre) [inline src/03-modality-agnostic-files.md:60]: This is the eventual article, it seems ? I'd expect that under "ReferencesAndLinks". I'd think to use this preregistration: https://osf.io/uf9gh

Comment (emdupre) [reply src/03-modality-agnostic-files.md:60]: I notice that the prereg itself doesn't have a DOI, which is maybe what influenced the decision ?

Still, I'd think we'd want to point to the original prereg in this field, if at all possible...

Comment (Remi-Gau) [reply src/03-modality-agnostic-files.md:60]: Oh dear. :facepalm: 
I did rush too much on this one. 

Thanks @emdupre for catching it. And yeah you are right this is exactly how I mixed things up: I assumed the DOI on the pre-reg page was that of the pre-reg and not the article.

**Comment by effigies** [comment]
@emdupre Does this look good to you?

**Comment by emdupre** [comment]
> @emdupre Does this look good to you?

I still have the same concerns that I expressed before, but the PR itself looks fine. Thanks for patching that link, @Remi-Gau !

**Comment by effigies** [comment]
Ah, I didn't scroll up very far and had forgotten that context.

It does seem like ReferencesAndLinks would be an appropriate place to put the pre-registration that already exists. If the goal is to indicate how to link to a pre-registration, would an example where we do cite one there work? Or is there an advantage to a new field?

**Comment by Remi-Gau** [comment]
> It does seem like ReferencesAndLinks would be an appropriate place to put the pre-registration that already exists. If the goal is to indicate how to link to a pre-registration, would an example where we do cite one there work? Or is there an advantage to a new field?

Hum... Gut feeling is that a link in the `ReferencesAndLinks` seems a bit "underspecifiied" for this. For a user you then would have to follow each link to know if it points to a paper or a pre-reg. Similarly the distinction would be lost for a machine, no?

NB: if we decide to put it in the `ReferenceAndLinks` remind me to revert the PR in the validator:
https://github.com/bids-standard/bids-validator/pull/1120

**Comment by effigies** [comment]
ReferencesAndLinks is entirely open, so you could say:

```
"ReferencesAndLinks": ["Study preregistration at https://blahblah"]
```

The question is whether having a specific preregistration field so that it could be directly parsed (as opposed to read by a human and understood) is that useful. I really don't have a strong opinion here.

**Comment by Remi-Gau** [comment]
> ReferencesAndLinks is entirely open, so you could say:
> 
> ```
> "ReferencesAndLinks": ["Study preregistration at https://blahblah"]
> ```

Oh. I was unclear about this. Good to know.


> The question is whether having a specific preregistration field so that it could be directly parsed (as opposed to read by a human and understood) is that useful. I really don't have a strong opinion here.

Hum... Part of me wants to "keep it simple" and say "Pre-reg are not THAT widespread in neuroiamging, so let's put in ReferenceAndLinks and cross that pre-reg bridge when we get to it".

The cautious part of me wants to have it in there early to prevent potentially having to change that later...

**Comment by emdupre** [comment]
> cautious part of me wants to have it in there early to prevent potentially having to change that later...

I'm not sure that I see how it would be backwards incompatible? Of course, I hope that by the time we get to BIDS 2.0 pre-registrations will be well-established in neuroimaging ! But I don't think we'd have to wait quite that long if it becomes a clear need in the community.

Maybe two questions: (1) do we know (approximately) how many datasets would use this now, and (2) is there a use case in mind for it being a unique field ?

I think those are the two doubts I still have. But I'd agree with @effigies that I don't have a really strong opinion, here !

**Comment by effigies** [comment]
> > cautious part of me wants to have it in there early to prevent potentially having to change that later...
> 
> I'm not sure that I see how it would be backwards incompatible?

Yeah, it wouldn't be backwards incompatible so much as something you need to know and care enough about to promote an entry in a references list to a preregistration field. Which is a concern, but not a change from the status quo.

In my estimation, for it to be actually useful, we would need tools that can make use of links from datasets to preregistrations OR a database of preregistered studies such that generic database-linking tools might connect the dots. In the absence of either, perhaps this is premature, since we also don't know what form of link would be most useful for such tools/databases. Maybe a URI would be less useful than a more structured object.

**Comment by Remi-Gau** [comment]
Yeah I tend to agree and the more I think about it the more it feels like this is a 20% case at the moment.

Will think about it a bit more but will most likely end up closing this PR.

> In my estimation, for it to be actually useful, we would need tools that can make use of links from datasets to preregistrations OR a database of preregistered studies such that generic database-linking tools might connect the dots.

For the record, OSF could provide the infrastructure to help create and host a [registry](https://osf.io/registries) to centralize all the pre-reg of certain "scientific community", but it needs the backing of a science organization. So as long as we don't get the higher spheres involved, this avenue might be complicated.

**Comment by Remi-Gau** [comment]
Will open a PR on the validator to revert the change this introduced.

**Comment by sappelhoff** [comment]
Thanks @Remi-Gau

**Pull request closed without merging**